File: backend/app/schemas/company.py
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, Field


class CompanyCreate(BaseModel):
    company_code: str = Field(..., min_length=1, max_length=100)
    company_name: str = Field(..., min_length=1, max_length=255)
    district: str = Field(..., min_length=1, max_length=255)
    province: str = Field(..., min_length=1, max_length=255)


class CompanyUpdate(BaseModel):
    company_code: str | None = Field(None, min_length=1, max_length=100)
    company_name: str | None = Field(None, min_length=1, max_length=255)
    district: str | None = Field(None, min_length=1, max_length=255)
    province: str | None = Field(None, min_length=1, max_length=255)


class CompanyOut(BaseModel):
    id: str
    company_code: str
    company_name: str
    district: str
    province: str
    created_at: datetime
    total_farms: int | None = 0
    total_zones: int | None = 0
    total_trees: int | None = 0

File: frontend/src/types/company.ts
export interface Company {
  _id: string; // MongoDB document identifier
  company_code: string;
  company_name: string;
  district: string;
  province: string;
  created_at?: string;
  total_farms?: number;
  total_zones?: number;
  total_trees?: number;
}

export type CreateCompanyRequest = Omit<Company, "_id" | "created_at" | "total_farms" | "total_zones" | "total_trees">;
export type UpdateCompanyRequest = Partial<CreateCompanyRequest>;
